### PR TITLE
Remove prefill logic from resource pool

### PR DESCRIPTION
**Pull request by harshit-gangal**
## Description

This PR modifies the resource pool by removing prefill logic and modifying benchmark tests.

## Related Issue(s)

- #9706 

## Checklist

-   [ ] "Backport me!" label has been added if this change should be backported
-   [ ] Tests were added or are not required
-   [ ] Documentation was added or is not required

### CODE REVIEW

**Comment by rsajwani** [comment]
I went through RFC as well but not able to find if this change is part of proposal #1, 2 or 3 .. Can you give me more detail why removing pre-fill logic

**Comment by harshit-gangal** [comment]
> I went through RFC as well but not able to find if this change is part of proposal #1, 2 or 3 .. Can you give me more detail why removing pre-fill logic

It is refactor of the pool, and not directly related to RFC.

The prefill logic was not used in vitess deployment as the default value is `0` and the connection pool should be having lazy behaviour regarding opening connections to mysql.

File: go/pools/resource_pool.go [hunk 1]
```diff
@@ -238,21 +206,13 @@ func (rp *ResourcePool) reopen() {
 // it will wait till the next resource becomes available or a timeout.
 // A timeout of 0 is an indefinite wait.
 func (rp *ResourcePool) Get(ctx context.Context) (resource Resource, err error) {
-	span, ctx := trace.NewSpan(ctx, "ResourcePool.Get")
```
Comment (systay) [inline go/pools/resource_pool.go:241]: this seems unrelated. Why remove the tracing here?

Comment (harshit-gangal) [reply go/pools/resource_pool.go:241]: The trace is already happening at one higher up level in connpool.Pool
Seems like a unnecessary trace.

```
	span.Annotate("capacity", p.Capacity())
	span.Annotate("in_use", p.InUse())
	span.Annotate("available", p.Available())
	span.Annotate("active", p.Active())

	if cp.timeout != 0 {
		var cancel context.CancelFunc
		ctx, cancel = context.WithTimeout(ctx, cp.timeout)
		defer cancel()
	}
	r, err := p.Get(ctx, nil)
```

**Review by systay — APPROVED** [review]
